src/Compiler/CompilerUtils.cpp

```diff
@@ -1060,21 +1066,16 @@ int compileModule(mlir::OwningOpRef<ModuleOp> &module,
     mlir::MLIRContext &context, std::string outputBaseName,
     EmissionTargetType emissionTarget) {
   setupModule(module, context, outputBaseName);
-
   mlir::PassManager pm(&context, mlir::OpPassManager::Nesting::Implicit);
-  // Initialize accelerator if required
-  if (acceleratorTarget.compare("") != 0) {
-    std::vector<accel::Accelerator *> *accTargets =
-        accel::Accelerator::getAcceleratorList();
-    assert(accTargets && "should not be null");
-
-    for (auto accel : *accTargets) {
-      if (accel->isActive()) {
-        accel->prepareAccelerator(module, context, pm, emissionTarget);
-      }
-    }
+
+  // Initialize accelerators if required.
```

Comment:
Code above did not work, and gave segfault. New code uses a factory method that "knows" how to create a accellerator instance based on  the `accelKind` enum.  This way we will not have to change this code when new accelerators are added.